frontend/src/metabase/query_builder/components/AlertModals.jsx

```diff
@@ -113,21 +116,28 @@ export class CreateAlertModalContent extends Component {
   onAlertChange = alert => this.setState({ alert });
 
   onCreateAlert = async () => {
-    const { createAlert, apiUpdateQuestion, onAlertCreated } = this.props;
+    const {
+      question,
+      createAlert,
+      apiUpdateQuestion,
+      updateUrl,
+      onAlertCreated,
+    } = this.props;
     const { alert } = this.state;
 
-    // Resave the question here (for persisting the x/y axes; see #6749)
-    await apiUpdateQuestion();
+    try {
+      this.setState({ formError: null });
 
-    await createAlert(alert);
+      await apiUpdateQuestion(question);
+      await createAlert(alert);
+      await updateUrl(question.card(), { dirty: false });
```

Comment:
It's important to change the URL only after the alert has been created, otherwise in case of error it won't be visible to the user.